dojo_toolkit/test_runner.py

```diff
@@ -2,59 +2,49 @@
 Module for running tests like doctest and unittest
 """
 import os
-from subprocess import PIPE, Popen, call
+import subprocess
 
 from .notifier import notifier
 
 
-class SubprocessTestRunner:
-    """
-    Base class to all test runners that use subprocess module.
-    """
-    cmd = None
-
+class DoctestTestRunner:
     def __init__(self, code_path, sound_player):
         self.code_path = code_path
         self.sound_player = sound_player
 
     def run(self):
-        """
-        run a test cmd using subprocess
-        """
-        process = Popen(
-            [self.cmd, self.code_path],
-            shell=True,
-            stdout=PIPE,
-            universal_newlines=True,
+        result = self._run_doctest()
+        self._clear_screen()
+        self._handle_result(result['is_success'])
+        print(result['output'])
+        return result['is_success']
+
+    def _run_doctest(self):
+        result = subprocess.run(
+            ["python", "-m", "doctest", self.code_path],
+            capture_output=True
```

Comment:
```suggestion
            ["python -m doctest " + self.code_path + "/*.py"],
            capture_output=True,
            shell=True,
            encoding="utf-8"
```

Reply:
The run args is now a string to work with the `*` when passing the code_path (which is the folder with the code inside)
`shell=True` must be True to be able to use the wildcard used in `*.py`
`encoding="utf-8"` automatically converts the output to utf-8